### DISCUSSION TITLE
How we can use typia gRPC functionality to communicate with two a simple node js server ?

### DISCUSSION THREAD

**Discussion by muhad-bk** [samchon/typia#1221] [Q&A]
Adding a simple example or blog post will help lots.  My preference is using `hyper express` or `uwebsockets.js` for the server.

**Comment by samchon** [comment]
@{number=1221; body=Typia does not provide a gRPC server/client transport layer. The protobuf surface is lower-level: it can generate `.proto` text with `typia.protobuf.message<T>()` and encode/decode `Uint8Array` payloads with `typia.protobuf.*Encode` / `typia.protobuf.*Decode`.

For two Node.js services, use your HTTP/WebSocket/gRPC framework for the transport, and put typia at the payload boundary:

```ts
// producer
const bytes = typia.protobuf.assertEncode<Message>(message);

// consumer
const message = typia.protobuf.assertDecode<Message>(bytes);
```

If another language or a gRPC toolchain must participate, export the schema with `typia.protobuf.message<Message>()` and feed that `.proto` into the normal protobuf/gRPC tooling. Hyper Express or uWebSockets.js can also transport the binary body directly; typia still only handles the schema and binary payload conversion.

Closing this as answered; no typia-side gRPC transport example is planned.}.body
